src/Http/Controllers/HealthCheckController.php

```diff
@@ -14,13 +16,31 @@ class HealthCheckController
      */
     private $health_check_service;
 
-    public function __construct(HealthCheckService $health_check_service)
+    /**
+     * @var Repository
+     */
+    private $config;
+
+    /**
+     * @param HealthCheckService $health_check_service
+     * @param Repository $config
+     */
+    public function __construct(HealthCheckService $health_check_service, Repository $config)
     {
         $this->health_check_service = $health_check_service;
+        $this->config = $config;
     }
 
-    public function checkHealth(): JsonResponse
+    /**
+     * @param Request $request
+     * @return JsonResponse
+     */
+    public function checkHealth(Request $request): JsonResponse
     {
+        $access_token = $this->config->get('health-check.access_token');
+        if ($access_token !== null && $access_token !== $request->query('token')) {
+            return new JsonResponse(null, Response::HTTP_NOT_FOUND);
```

Comment:
I'm not really convinced about security by obscurity. I get that this keeps bad guys away, but can be a nightmare for developers if they make a mistake with the token. I personally will prefer a `403` instead, that way it's clear the health check is up and running.

Reply:
My intention was to prevent brute force, but then again it's an assumption someone would stop anyway.

Reply:
changed.